js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetExecutionButton.tsx

```diff
@@ -128,6 +128,38 @@ export const ERROR_INVALID_ASSET_SELECTION =
   ` the same code location and share a partition space, or form a connected` +
   ` graph in which root assets share the same partitioning.`;
 
+function materializationDisabledReason(all: Asset[], materializable: Asset[]) {
+  if (!all.length) {
+    return 'Select one or more assets to materialize';
+  }
+  if (all.some((a) => !a.hasMaterializePermission)) {
+    return 'You do not have permission to materialize assets';
+  }
+  if (all.every((a) => !a.isExecutable)) {
+    return 'External assets cannot be materialized';
+  }
+  if (all.every((a) => a.isObservable)) {
+    return 'Observable assets cannot be materialized';
+  }
+  if (materializable.length === 0) {
+    return 'No executable assets selected.';
+  }
+  return null;
+}
+
+function observationDisabledReason(all: Asset[], observable: Asset[]) {
+  if (!all.length) {
+    return 'Select one or more assets to observe';
+  }
+  if (all.some((a) => !a.hasMaterializePermission)) {
+    return 'You do not have permission to observe assets';
+  }
+  if (observable.length === 0) {
+    return 'Assets do not have observation functions';
+  }
+  return null;
+}
```

Comment:
Might be worth writing some simple jest specs for these.

Reply:
Yeah good call I'll put one together